mantidproject/mantid
# Set application name for the UsageService on workbench startup

~~**Requires #23875 to be merged first**~~

**Description of work.**

Sets the application name for the usage service to report on startup of the workbench.

**Report to:** [nobody]. 

**To test:**

Visit https//webhook.site and copy the unique url it creates for you (don't close the page). This will be used to capture the usage reports rather than putting them in the production database.

Edit the `Mantid.user.properties` file and set
```
usagereports.enabled = 1
errorreports.rooturl = <webhook-url>
usagereports.rooturl = <webhook-url>
```
where `<webhook-url>` should be replaced with the url copied from the webhook page.

Start the workbench and when it has started go back to the page webhook page in the browser and see that the request has been captured there. The application name should be `mantidworkbench` and **not** `python`  

*This does not require release notes* because **the workbench is not released yet**

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
